### ISSUE TITLE
Exoscale: set NetworkManager addr-gen-mode to eui64

### ISSUE THREAD

**Issue by irregulator** [coreos/fedora-coreos-tracker#513]
Hello and thanks for maintaining Fedora CoreOS.

This is an issue related to Exoscale Fedora CoreOS image and affects IPv6 connectivity of Exoscale's Cloud platform guests.

### Problem statement

Exoscale's cloud platform expects the the Fedora CoreOS guest will use the information included in a IPv6 router advertisement and its hardware address (MAC) to assign EUI64 global scope and link-local scope IPv6 addresses to its interface. Using EUI64 means that the guest will have a predictable set of addresses and the hypervisor will be able to normally route traffic to it.

NetworkManager, used in Fedora CoreOS guests, defines `addr-gen-mode=stable-privacy` for its default connection:
```
[root@VM-a7ede26d-3d7b-40ac-82ac-78d6b33615af core]# grep -iFR addr-gen-mode /etc/NetworkManager/
/etc/NetworkManager/system-connections/default_connection.nmconnection:addr-gen-mode=stable-privacy
```

This means that the IPv6 addresses assigned to guest's interface will not be predictable and thus the hypervisor cannot route IPv6 traffic to guest.

### Proposed fix

To fix the issue we need to set NetworkManager's property `addr-gen-mode` to `eui64` for the Exoscale Fedora CoreOS image. Please let us know what's the most appropriate place/way to set this value for Exoscale image and we shall handle a PR.

### rpm-ostree status
```
[root@VM-a7ede26d-3d7b-40ac-82ac-78d6b33615af core]# rpm-ostree status
State: idle
Deployments:
● ostree://fedora:fedora/x86_64/coreos/testing
                   Version: 31.20200517.2.0 (2020-05-19T10:24:32Z)
                    Commit: 5c3f8198e72a05adab4eb7d087ef3a625008dcc73bd0416a444cdce084278587
              GPGSignature: Valid signature by 7D22D5867F2A4236474BF7B850CB390B3C3359C4
```

**Comment by cgwalters** [comment]
Hi, thanks for filing this issue!

Offhand, I don't see an issue changing the IPv6 mode to be predictable in all cloud environments by default - there's no privacy concerns there.

Or perhaps said more strongly: we only want to use `stable-privacy` on bare metal.  

The implementation of this policy though is somewhat tricky because a central thing for Fedora CoreOS is that while we have a different bootimage per platform, there is exactly one update stream that applies across all of them.

This came up in e.g. [custom NTP configuration for Azure](https://github.com/coreos/fedora-coreos-config/pull/367) where we wrote a systemd generator which keys off the `ignition.platform.id` kernel argument.

I could imagine doing something like this for this too...would need to investigate whether we can dynamically reconfigure NetworkManager in this way at runtime (e.g. via dropping a file in `/run/NetworkManager` or so).

**Comment by lucab** [comment]
Gah, I started looking into this a while ago but forgot to record feedback.

> Offhand, I don't see an issue changing the IPv6 mode to be predictable in all cloud environments by default - there's no privacy concerns there.
> Or perhaps said more strongly: we only want to use `stable-privacy` on bare metal.

My reading of https://tools.ietf.org/html/rfc7217 is that this is mostly meant at avoiding tracking via MAC addresses for clients moving across network prefixes.
For CoreOS nodes I'm not sure whether this setting really helps, as the hosts do not (generally) move across networks and this only affects SLAAC addresses (i.e. bare-metal users with stronger requirements will define their own DHCPv6 topology anyway).

As such, I'm thus wondering whether we should stick to uniform defaults and state either that:
 1. we are fine with EUI64 everywhere (including bare-metal) as it helps automating SLAAC-based networks, and there are no real privacy concerns for our specific usecase
 2. we have real privacy concerns for CoreOS hosts, thus we should work with Exoscale to come to a solution which works with SLAAC and RFC7217

**Comment by irregulator** [comment]
Hi!

It seems the originally reported behavior has been changed in Fedora Core OS version 32:

```
[root@VM-95372ecc-f91e-4f45-a1ac-ebc69388c55b ~]# grep -iFR eui64 /etc/NetworkManager/
/etc/NetworkManager/system-connections/default_connection.nmconnection:addr-gen-mode=eui64


[root@VM-95372ecc-f91e-4f45-a1ac-ebc69388c55b ~]# rpm-ostree status
State: idle
Deployments:
● ostree://fedora:fedora/x86_64/coreos/stable
                   Version: 32.20200615.3.0 (2020-06-30T09:14:54Z)
                    Commit: abb4b978999bd689ff36114e99c737be16026bc154a4428d4c84a340758c5b01
              GPGSignature: Valid signature by 97A1AE57C3A2372CCA3A4ABA6C13026D12C944D0

  ostree://fedora:fedora/x86_64/coreos/stable
                   Version: 32.20200601.3.0 (2020-06-16T08:52:21Z)
                    Commit: b51037798e93e5aae5123633fb596c80ddf30302b5110b0581900dbc5b2f0d24
              GPGSignature: Valid signature by 97A1AE57C3A2372CCA3A4ABA6C13026D12C944D0
```

`addr-gen-mode` is `eui64` so SLAAC and IPv6 connectivity works as expected for instances in Exoscale cloud. So you may close current issue. Thanks for maintaining CoreOS! :slightly_smiling_face:

**Comment by brtknr** [comment]
I wonder if there has been a regression as this doesn't appear to be the case anymore.

**Comment by brtknr** [comment]
Yep its gone back to `stable-privacy`:
```
nmcli> print ipv6.addr-gen-mode
ipv6.addr-gen-mode: stable-privacy
```

**Comment by dustymabe** [comment]
I think the change in behavior is a side effect of a few other things. We used to carry forward the initrd generated config even if it was completely default (which is why this issue was working fine in F32+). The initrd generated config doesn't seem to have `ipv6.addr-gen-mode=stable-privacy`. 

We stopped carrying forward the initrd generated config in the "pure defaults" case in https://github.com/coreos/fedora-coreos-config/pull/773 (for explanation see https://github.com/coreos/fedora-coreos-tracker/issues/696). So this was a purely unintentional change to begin with, and unluckily we reverted it.

Either way we'll try to see what we want the behavior to be and if we can set this at a higher level. See https://github.com/coreos/fedora-coreos-tracker/issues/907#issuecomment-886817668
